pkg/agent/core/ngt/service/ngt.go

```diff
@@ -135,9 +135,19 @@ func New(cfg *config.NGT, opts ...Option) (nn NGT, err error) {
 		}
 	}
 
-	n.kvs = kvs.New()
 	n.dim = cfg.Dimension
 
+	n.kvs = kvs.New(func() []kvs.Option {
+		if cfg.KVSDB.Concurrency < 1 {
+			return []kvs.Option{
+				kvs.WithErrGroup(n.eg),
```

Comment:
```suggestion
				// n.eg is global errgroup we can use it only concurrency < 1 (which means unlimited)
				// if we use global errgroup under the limited concurrency it will affect other daemon process
				kvs.WithErrGroup(n.eg),
```